### PR TITLE
Visibility-v2: use IntersectionObserverPolyfill

### PR DESCRIPTION
**Pull request by lannka**
Part of #6828

### CODE REVIEW

File: extensions/amp-analytics/0.1/visibility-impl.js [hunk 1]
```diff
@@ -354,13 +361,23 @@ export class Visibility {
...
+        entries.forEach(this.onIntersectionChange_.bind(this));
+      };
+
+      if (nativeIntersectionObserverSupported(this.ampdoc.win)) {
+        this.intersectionObserver_ = new this.ampdoc.win.IntersectionObserver(
+            onIntersectionChanges, {threshold: DEFAULT_THRESHOLD});
+      } else {
+        this.intersectionObserver_ = new IntersectionObserverPolyfill(
+            onIntersectionChanges, {threshold: DEFAULT_THRESHOLD});
+        const viewport = viewportForDoc(this.ampdoc);
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/visibility-impl.js]: probably add a TODO before, that we will eventually remove the ticker.

File: extensions/amp-analytics/0.1/test/test-visibility-impl.js [hunk 2]
```diff
@@ -507,13 +505,24 @@ describe('amp-analytics.visibility', () => {
...
       callbackSpy2 = sandbox.stub();
-      sandbox.stub(ampdoc.win, 'IntersectionObserver', callback => {
-        inObCallback = callback;
-        return {
-          observe: observeSpy,
-          unobserve: unobserveSpy,
-        };
-      });
+
+      if (inob.nativeIntersectionObserverSupported(ampdoc.win)) {
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/test/test-visibility-impl.js]: During local test, I believe it is always the case that nativeIntersectionObserver is supported. Also I fail to find where the usage of IntersectionObserver logic was tested, don't quite understand why are we having this https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/test/test-visibility-impl.js#L116

File: extensions/amp-analytics/0.1/visibility-impl.js (hunk 1 above)
Comment (lannka) [inline extensions/amp-analytics/0.1/visibility-impl.js]: Done

File: extensions/amp-analytics/0.1/test/test-visibility-impl.js (hunk 2 above)
Comment (lannka) [inline extensions/amp-analytics/0.1/test/test-visibility-impl.js]: splitted tests .

Comment (zhouyx) [inline extensions/amp-analytics/0.1/test/test-visibility-impl.js]: OK.... I think we will eventually need an integration test though 😢

**Pull request merged**